pkg/plugins/apiserver/server.go

```diff
@@ -0,0 +1,217 @@
+package apiserver
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/authtoken"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/live"
+	"github.com/grafana/grafana/pkg/setting"
+
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/server"
+	liveDto "github.com/grafana/grafana-plugin-sdk-go/live"
+	grpcAuth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	logger = log.New("plugin_grpc_api")
+)
+
+func ProvideService(
+	cfg *setting.Cfg,
+	GrafanaLive *live.GrafanaLive,
+	DatasourceCache datasources.CacheService,
+	JWT *authtoken.JWT,
+) (*GRPCAPIServer, error) {
+	s := &GRPCAPIServer{
+		Cfg:             cfg,
+		GrafanaLive:     GrafanaLive,
+		DatasourceCache: DatasourceCache,
+		JWT:             JWT,
+	}
+
+	logger.Info("Plugin GRPC API server initialization", "address", s.Cfg.GRPCServerAddress, "network", s.Cfg.GRPCServerNetwork, "tls", s.Cfg.GRPCServerUseTLS)
```

Comment:
This is a great start at a GRPC server in grafana -- however it is written so it is scoped for just plugins.  Do you think it makes sense so it is scoped at the root, but still accessible from plugins?